Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM debian:stretch
+
+#TODO: Verify Go version for mender 2.0.0 beta release
+ARG MENDER_VERSION=2.0.0-beta
+ARG GOLANG_VERSION=1.11.5
+
+RUN apt-get update && apt-get install -y \
+    git wget \
+    build-essential debhelper devscripts \
+    liblzma-dev
+
+# TODO: test with gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf.tar.xz !!
+RUN git clone --progress https://github.com/raspberrypi/tools.git
+ENV CC "arm-linux-gnueabihf-gcc"
+ENV PATH "$PATH:/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin"
```

Comment:
As already @mirzak noticed, why do you need raspberrypi tools at all? Is it only needed to get a toolchain to build Mender Client? If so, then you'd better take some more recent toolchain from e.g. linaro or even simply try to install it from packages: gcc-arm-linux-gnueabihf. 
Please take a look how both approaches are done in mender-convert: https://github.com/mendersoftware/mender-convert/commit/a231c92edde3c01d7902dde03772bad60e2fbc5b#diff-3254677a7917c6c01f55212f86c57fbfL38

Reply:
Exactly my comment as well. You can try `apt install crossbuild-essential-armhf` perhaps.

Reply:
I discussed it with @mirzak this morning.

Turns out that the problem is that my simulated Raspberry Pi is using ARMv6, so that when I tested with `gcc-arm-linux-gnueabihf` didn't work. But now I am using a real Raspberry Pi 3 (ARMv7) and can use `gcc-arm-linux-gnueabihf` :+1:

Reply:
You should compile as ARMv5 though, because it is the most compatible. There are options for gcc to turn this on. This package may be installed on a variety of devices, so we should not assume ARMv7.

@mirzak: Feel free to correct me if you know more about the market share of each architecture, but this is the approach I used for our GRUB binary.

Reply:
Discussed with @mirzak on chat.

Turns out that `gcc-arm-linux-gnueabihf` does only support ARMv7. Also Debian upstream seems to support only v7 and newer.

I tried with GCC flag `-march=armv5te` but the toolchain seemed to ignore it. When manually compiling a hello world program with the flag, I get an error saying that it is  unimplemented.

So my suggestion is to leave gcc compilation for ARMv7.

Reply:
Ok, then we have no choice, I guess.